### ISSUE TITLE
Game crashes when looking at rendered landscape?

### ISSUE THREAD

**Issue by Tiepilot1138** [tsunamayo/Skywanderers#149]
My game crashes every time I turn and face the ring (landscape).  I only see half of the ring rendered.  As soon as it enters my view the game freezes and crashes.  It does this on both PC's that I have been able to get it to start on.  This occurs right after the world loads and I turn my view in that direction.

Machine specs:

AMD FX-8120 eight core 4.1 ghz
Win 10 64 bit
16 gig ram
GeForce GTX 550 Ti

Updated drivers, still crashes

[output_log.txt](https://github.com/tsunamayo/Skywanderers/files/1412302/output_log.txt)

![sw ring](https://user-images.githubusercontent.com/33068492/31964689-d35b1dee-b8d2-11e7-8cac-c2c3c621ca49.jpg)
![20171024135650_1](https://user-images.githubusercontent.com/33068492/31964769-1d1245d4-b8d3-11e7-9e7d-f6b3e54365f3.jpg)

**Comment by tsunamayo** [comment]
Hello, what is the GFX card on the second config? You are running the game just below the minimum configuration - but I am still not 100% sure this is because of that yet.
Could you please go to the steam installation folder and check whether the following file exists?
\Skywanderers_Data\Plugins\AsyncTextureReader.dll
(In steam right click on game, then Property, then Local Files, then Browse Local files)
Thanks!

**Comment by Tiepilot1138** [comment]
This machine has AsyncTextureReader.dll in the specified folder.

The other machine is a Dell XPS 15 9550 with a GeForce GTX 960M.

**Comment by Tiepilot1138** [comment]
I noticed something else that didn't occur to me yesterday.  Using the new build just now, on the XPS, I realized that it locks up when I look at the right side of the ring.  I tried rotating right and left at the start point (I walked out the door and turned around first).  When turning to the right, I can see most of the ring (that renders, anyway), but when it gets to the right side of the ring it freezes.  When turning to the left, the ring barely comes into view before freezing.  I've also noted it does it if I am inside and can't see the ring at all, I just have to be facing in that direction,  The two pics below show how far it goes when it freezes.  Other than this, it seems to run fine (as far as I've been able to go, anyway).  I did go out to the tutorial building and was able to interact (am I supposed to be on the ring at this point?)  But every time I turn and face that direction it freezes and crashes.

![20171025_100310](https://user-images.githubusercontent.com/33068492/32003337-b5ad1b90-b96c-11e7-82b2-8358b1d057f2.jpg)
![20171025_100618](https://user-images.githubusercontent.com/33068492/32003341-b8074618-b96c-11e7-8749-b2889afe5e83.jpg)

**Comment by AssIn9** [comment]
[b]Description of issue:[/b] Same as above. 

[b]Frequency of issue:[/b] every time i face that dirrection

[b]Steps to replicate issue:[/b] load into game and turn 180. 

[b]Hardware Configuration:[/b] //FOR CRASHES ONLY//
OS: win 10 pro
CPU: i7-4770
GPU:gtx 780 ti
RAM:16.0gb
Motherboard/laptop reference: Gigaabyte Tech...  z87m-d3h

[output_log.txt](https://github.com/tsunamayo/Skywanderers/files/1416393/output_log.txt)

**Comment by AssIn9** [comment]
Able to play if i delete torus.swbp

**Comment by Tiepilot1138** [comment]
AssIn9, I just found that if I delete torus.swbp I can play too.  Unfortunately that also deletes all the tutorial stuff...  :(

**Comment by tsunamayo** [comment]
Yes you are clearly having trouble with the voxel layer of the Torus. Your second card should be powerfull enough. Could you please check your GFX drivers?

**Comment by AssIn9** [comment]
I just updated my video card drivers to the latest, Same result.

![image](https://user-images.githubusercontent.com/6031455/32067164-3c995d1a-ba50-11e7-956f-53f5b3c8610e.png)

**Comment by tsunamayo** [comment]
Could you please unzip the following file under 
C:\Users\{YOUR USER]\AppData\LocalLow\TsunamayoCorp\Skywanderers\Save_Data\Characters

[Characters.zip](https://github.com/tsunamayo/Skywanderers/files/1417378/Characters.zip)

Could you restore back the the Torus.swbp first.

Could you please do the following:
1) Load Empty_world : is it working? Do you see the galaxy background?
2) Load Torus: is it working? Do you see the torus
3) if 2) is yes, could you please hit ] key a few time to move faster (but not too fast), and then try move slowly toward the torus until you reach the ground. Is it working or crashing?
4) could you load Asteroids, then tell me if you see the asteroids.

Thanks

**Comment by Tiepilot1138** [comment]
All are successes.  
1. Saw the stars (so pretty...)
2.  Saw the torus, was able to move to it and passed right through it (was that supposed to happen?)
3.  Saw asteroids, and was able to move around them.  

Oddly, this happened when I passed into the torus.
![20171026212821_1](https://user-images.githubusercontent.com/33068492/32084215-cb549228-ba95-11e7-8e81-1c60fdbd2114.jpg)
![20171026212840_1](https://user-images.githubusercontent.com/33068492/32084219-d037a064-ba95-11e7-85b4-9c953c7304e4.jpg)
![20171026212910_1](https://user-images.githubusercontent.com/33068492/32084222-d202a696-ba95-11e7-8261-4b73bb1c3c87.jpg)
![20171026212934_1](https://user-images.githubusercontent.com/33068492/32084223-d47f8858-ba95-11e7-8906-f9fad8a0e8df.jpg)

**Comment by Tiepilot1138** [comment]
I forgot to mention that when I first restored the torus, it was still only half visible.  The update did not correct that.

**Comment by AssIn9** [comment]
I don't seem to have any zip files in characters. so, i renamed the file://C:\users\%usermane%\appdata\locallow\tsunamayocorp\skywanderers\ folder to oldskywanderers and restarted the game after a full delete and reload from steam with the same results. 

if i delete the Taurus bp from C:\Program Files (x86)\Steam\steamapps\common\Skywanderers\Skywanderers_Data\StreamingAssets\PlayerEntities the game plays fine. 
Thank you

**Comment by Tiepilot1138** [comment]
I also noticed that when starting a fresh game (after a reinstall) I can walk through the walls of the starting building.  Is that supposed to happen?  This is with the original torus that locks the game.

**Comment by AssIn9** [comment]
Fresh load of windows with fresh load of steam with fresh load of SW. 

[output_log.txt](https://github.com/tsunamayo/Skywanderers/files/1424341/output_log.txt)

still crashes when looking at ring. 

ran and didn't look at the ring. then closed. SW was using 25GB ram.  and 81% of my cpu allocated
![image](https://user-images.githubusercontent.com/6031455/32135580-73e0e4d4-bbcf-11e7-8e79-a29abba28165.png)

**Comment by AssIn9** [comment]
I cranked my speed up, and moved to the asteroid belt. 
when i turned around, i didn't crash immediately. I approached the ring unitl it chrashed my game. 
![image](https://user-images.githubusercontent.com/6031455/32135815-0b58025e-bbd3-11e7-9968-60ff81aadb86.png)
[output_log.txt](https://github.com/tsunamayo/Skywanderers/files/1424368/output_log.txt)

**Comment by Tiepilot1138** [comment]
I got the game running on a freshly imaged Surface Pro 4, and have the same ring issue.  Game seems to work fine once I delete torus.swbp.  That's 3 for 3 on my systems.

**Comment by tsunamayo** [comment]
@AssIn9 sorry can you check if your GFX card configuration it seems that you are running it on a 256mb video card (so I guess an integrated one). Thanks

**Comment by CyaNox** [comment]
In #317 I found that there is a dependency lacking for the MSVC Runtime 2015 as per: https://github.com/tsunamayo/Skywanderers/issues/317#issuecomment-344861361

You can try downloading and installing it from https://www.microsoft.com/en-gb/download/details.aspx?id=48145 and see if that helps.

**Comment by AssIn9** [comment]
@tsunamayo I only have one video device installed. and it is at default settings from nvidia. 
![image](https://user-images.githubusercontent.com/6031455/32888593-c4b8f480-ca95-11e7-85e8-60222e01e4d4.png)

**Comment by AssIn9** [comment]
the msvc runtime 2015 seems to fix it for me. will test another computer aswell.

**Comment by Tiepilot1138** [comment]
I was able to test this on my surface 4 and dell xps15, and it worked.  The ring is there and doesn't crash the game.  I had to run the executable on the surface, but the xps said it was already there.  I verified it worked on both (it did not work on either the other day).  I will test on my desktop tonight as well.  Oddly, I did notice that when I jumped into the hovercar for my first joyus ride around the torus (see what I did there?), the hovercraft drove straight and level, which made it pass below the surface of the ring.  Also, the yaw controls were backwards.  This fix should be stickied somewhere.  Thanks for the effort, guys!
